app/src/main/java/com/worldsnas/starplayer/ExoPlayerService.kt

```diff
@@ -0,0 +1,387 @@
+            }
+            mediaSource = ConcatenatingMediaSource(*mediaSources)
+            player.prepare(mediaSource)
+            startPlayer()
+        }
+    }
+
+
+    private fun pausePlayer() {
+        playbackPosition = player.currentPosition
+        currentMusicPosition = player.currentWindowIndex
+        player.playWhenReady = false
+        showCurrentTrackNotification(musicList[player.currentWindowIndex])
+    }
+
+    private fun startPlayer() {
+        player.seekTo(currentMusicPosition, playbackPosition)
+        player.playWhenReady = true
+        showCurrentTrackNotification(musicList[player.currentWindowIndex])
+    }
+
+    private fun initializePlayer() {
+        player = SimpleExoPlayer.Builder(this).build()
+        player.setAudioAttributes(
+            com.google.android.exoplayer2.audio.AudioAttributes.Builder()
+                .setUsage(C.USAGE_MEDIA)
+                .setContentType(C.CONTENT_TYPE_SPEECH)
+                .build(), true
+        )
+
+
+        player.addListener(this)
+
+    }
+
+    private fun buildMediaSource(uri: Uri): MediaSource {
+        val userAgent = "star_player"
+        return ProgressiveMediaSource.Factory(DefaultHttpDataSourceFactory(userAgent))
+            .createMediaSource(uri)
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        initializePlayer()
+    }
+
+    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+
+        when (intent.action) {
+            PLAY_ACTION -> {
+
+                startPlayer()
+            }
+            PAUSE_ACTION -> pausePlayer()
+            START_TRACKS_ARRAY_ACTION -> {
+                musicList = intent.getParcelableArrayListExtra(
+                    TAG
+                )!!
+                currentMusicPosition = intent.getIntExtra(CURRENT_MUSIC_POSITION_KEY, 0)
+                play(
+                    musicList
+                )
+            }
+            START_TRACK_ACTION -> play(
+                intent.getParcelableExtra<Music>(
+                    TAG
+                )!!
+            )
+
+            PREV_ACTION -> {
+                val prev = player.previousWindowIndex
+                if (prev >= 0 && prev <= mediaSource.size) {
+                    currentMusicPosition = prev
+                } else {
+                    currentMusicPosition = 0
+                    playbackPosition = 0
+                }
+                startPlayer()
+            }
+
+            NEXT_ACTION -> {
+                val next = player.nextWindowIndex
+                if (next >= 0 && next < mediaSource.size) {
+                    currentMusicPosition = next
+                } else {
+                    currentMusicPosition = 0
+                    playbackPosition = 0
+                }
+                startPlayer()
+            }
+            STOP_ACTION -> {
+                stopAction()
+            }
+            else -> stopAction()
```

Comment:
benazaram throw konim behtare
sari tar moshkelo peyda mikonim
(albate maghadir ro ham pas bedim be exception)

Reply:
dorostesh mikonam